my-app/src/frontend/pages/login/Login.js

```diff
@@ -0,0 +1,83 @@
+import "./login.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts/auth-context";
+import axios from "axios";
+
+export function Login() {
+  const { authState, authDispatch } = useAuth();
+  const { user, error } = authState;
+  const navigate = useNavigate();
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email: user.email,
+        password: user.password,
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("token", response.data.encodedToken);
+      authDispatch({ type: "TOKEN", payload: response.data.encodedToken });
+      navigate("/");
+    } catch (error) {
+      authDispatch({
+        type: "ERROR",
+        payload: "Wrong credentials,please try again",
+      });
+      console.log(error);
+    }
+  };
+
+  return (
+    <>
+      <form className="login-container" onSubmit={loginHandler}>
+        <h3>LOGIN</h3>
+        <div className="input email-label">
+          <div>
+            <label htmlFor="name" className="input-email-label">
+              Email
+            </label>
+          </div>
+          <div>
+            <input
+              type="email"
+              class="email-input user-input"
+              onChange={(e) =>
+                authDispatch({ type: "EMAIL", payload: e.target.value })
```

Comment:
handling form data with onChange can be done simply with useState without context since changes in formdata need not reflect elsewhere

Reply:
Only after form submission / submission function context could be used to maintain even data

Reply:
I completely agree with this. Please make the necessary changes.